racetrack_client/racetrack_client/utils/semver.py

```diff
@@ -125,26 +133,59 @@ def __hash__(self):
 
 
 class SemanticVersionPattern:
-    valid_pattern_regex = re.compile(r'(\d+|x)(\.(\d+|x))*')  # 1.2.x or 1.x or 1.x.x
+    # 1.2.x or 1.x or 1.x.x
+    x_pattern_regex = re.compile(r'(\d+|x)(\.(\d+|x))*')
+    # 1.2.* or 1.2.3-*-bullseye or 1.*.*
+    asterisk_pattern_regex = re.compile(r'(?P<major>\d+|\*)(?P<minor>\.(\d+|\*))?(?P<patch>\.(\d+|\*))?(?P<label>\-.+)?')
 
-    def __init__(self, pattern: str) -> None:
+    def __init__(self, regex_pattern: re.Pattern) -> None:
+        self.regex_pattern = regex_pattern
+
+    @classmethod
+    def from_x_pattern(cls, pattern: str) -> 'SemanticVersionPattern':
         """
         :param pattern: version pattern containing "x" wildcards, eg. "1.2.x", "2.x"
         """
-        match = self.valid_pattern_regex.fullmatch(pattern)
+        match = cls.x_pattern_regex.fullmatch(pattern)
         if not match:
             raise ValueError(f"Version pattern '{pattern}' is invalid, should be '1.x' or '1.2.x'")
         if 'x' not in pattern:
             raise ValueError(f"Version pattern '{pattern}' doesn't contain 'x' wildcard")
 
-        self.regex_pattern = re.compile(pattern.replace('.', r'\.').replace('x', r'.+'))
+        regex_pattern = re.compile(pattern.replace('.', r'\.').replace('x', r'.+'))
+        return SemanticVersionPattern(regex_pattern)
+
+    @classmethod
+    def from_asterisk_pattern(cls, pattern: str) -> 'SemanticVersionPattern':
+        """
+        :param pattern: version pattern containing "*" wildcards, eg. 1.2.* or 1.2.3-*-bullseye or 1.*.*
+        """
+        match = cls.asterisk_pattern_regex.fullmatch(pattern)
+        pattern2 = re.escape(pattern)
+        pattern2 = pattern2.replace('\\*', '.+')
+        if not match:
+            raise ValueError(f"Version pattern '{pattern}' is invalid, should be '1.2.*' or '1.2.3-*-bullseye' or '1.*.*'")
+        if '*' not in pattern:
+            raise ValueError(f"Version pattern '{pattern}' doesn't contain '*' wildcard")
+
+        regex_pattern = re.compile(pattern2)
+        return SemanticVersionPattern(regex_pattern)
 
     def matches(self, version: SemanticVersion) -> bool:
         return self.regex_pattern.fullmatch(str(version)) is not None
 
     @staticmethod
-    def is_wildcard_pattern(pattern: str) -> bool:
-        match = SemanticVersionPattern.valid_pattern_regex.fullmatch(pattern)
+    def is_x_wildcard_pattern(pattern: str) -> bool:
+        """Check whether it's a valid version pattern containing "x" wildcards"""
+        match = SemanticVersionPattern.x_pattern_regex.fullmatch(pattern)
         if not match:
             return False
         return 'x' in pattern
+
+    @staticmethod
+    def is_asterisk_wildcard_pattern(pattern: str) -> bool:
```

Comment:
Nitpick: maybe removing `wildcard` from the name of this function, and the one above, would make them less vocal, simpler and easier to understand. For me the word `wildcard` is almost synonymous with `?*` chars. So seeing `is_x_wildcard_pattern` I start thinking that this function looks for `x` or `*` chars. Maybe it's just me.. for consideration.